### PR TITLE
Require storage server running for uptime proofs

### PR DESCRIPTION
**Pull request by msgmaxim**

### CODE REVIEW

File: src/cryptonote_core/cryptonote_core.cpp [hunk 1]
```diff
@@ -2019,6 +2031,25 @@ namespace cryptonote
...
+
+    const auto ip = string_tools::get_ip_string_from_int32(m_sn_public_ip);
+    const auto port = std::to_string(m_storage_port);
+
+    http_client.set_server(ip, port, boost::none, epee::net_utils::ssl_support_t::e_ssl_support_disabled);
+
+    cryptonote::COMMAND_RPC_PING_STORAGE::request req = AUTO_VAL_INIT(req);
+    cryptonote::COMMAND_RPC_PING_STORAGE::response resp = AUTO_VAL_INIT(resp);
+
+    return net_utils::invoke_http_json_rpc("/json_rpc", "ping", req, resp, http_client, PING_TIMEOUT);
```
Comment (Doy-lee) [inline src/cryptonote_core/cryptonote_core.cpp]: Can this be simplified further by just using http_simple_client::connect()? We can get rid of the RPC boilerplate which is all noise in this instance.

Comment (Doy-lee) [reply src/cryptonote_core/cryptonote_core.cpp]: Hmm I see there's the equivalent GET response in the storage server now.

Comment (msgmaxim) [reply src/cryptonote_core/cryptonote_core.cpp]: I was thinking about that, but I wasn't sure if we wanted to use non-empty request/response or not.

**Comment by msgmaxim** [comment]
Closing this in favour of #631

**Pull request closed without merging**